src/TBuffer.cpp

```diff
@@ -2430,11 +2430,20 @@ TBuffer TBuffer::copy(QPoint& P1, QPoint& P2)
     if ((x < 0) || (x >= static_cast<int>(buffer.at(y).size())) || (P2.x() < 0) || (P2.x() > static_cast<int>(buffer.at(y).size()))) {
         x = 0;
     }
-
+    int linkId, oldLinkId, id;
```

Comment:
`oldLinkId` is not being initialised before it's use in the `if` test as far as I can tell, you need to set it to `0` I think.

Also, for simpler/cleaner code you should declare `id` *inside* the `for` loop and initialise it to `0` there - then you do not need the second `if` at all...!

Actually, there is no need to not move the declaration of `linkId` into inside the loop either, and assign the value to it at the same time.

:neckbeard:

Reply:
Hang on:
* `oldLinkId` must be initialised to *something* BEFORE it is compared to `linkId` for the first time.
* `linkId` will be set to *something* every time around the loop and since it is not used outside the loop there is no point in having it declared outside the loop.

I am assuming that `TChar::linkIndex()` returns `0` if there is no link value set for the particular `TChar` and that it will be set to the same non-zero value for each `TChar` in the string that has the particular link - which *seems* to be the case AFAICT.

Lets see if another set of 👀 can clarify things at all.. @vadi2 can you take a look at the logic / flow here?

Reply:
yes, you are correct and I probably should initialize linkId in the loop (will change it). 

There are still 2 (if) checks needed because it is also possible that there are 2 or more links directly beside each other so they are not interrupted by a linkIndex of 0.

So if 0 the link ends and if linkId != oldLinkId a new link starts.